### PR TITLE
Bugfix: undo-redo history at new created files

### PR DESCRIPTION
**Pull request by harshad1**
The undo redo tracker was not being set on opening an empty file / creating a new file.

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/frontend/textview/TextViewUtils.java [hunk 1]
```diff
@@ -713,39 +713,4 @@ public static Boolean isImeOpen(final View view) {
         }
         return null; // Uncertain
     }
-
-    public interface MTextWatcher extends TextWatcher {
-        // Classes inheriting from this class will be disabled in withAutoFormatDisabled
-    }
-
-    public static void withAutoFormatDisabled(final Editable editable, final GsCallback.a0 callback) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/frontend/textview/TextViewUtils.java:721]: Unnecessary and non-performant code. Switching back to the old method for simplicity.

File: app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java [hunk 2]
```diff
@@ -336,10 +336,15 @@ public void setAutoFormatEnabled(final boolean enable) {
 
     // Run some code with auto formatters and accessibility disabled
     public void withAutoFormatDisabled(final GsCallback.a0 callback) {
+        final boolean enabled = getAutoFormatEnabled();
         try {
             _accessibilityEnabled = false;
-            TextViewUtils.withAutoFormatDisabled(getText(), callback);
+            if (enabled) {
+                setAutoFormatEnabled(false);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java:343]: Much simpler and easier to follow

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java [hunk 3]
```diff
@@ -186,6 +186,11 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         _document.resetChangeTracking(); // force next reload
         loadDocument();
 
+        // If not set by loadDocument, se the undo-redo helper here
+        if (_editTextUndoRedoHelper == null) {
+            _editTextUndoRedoHelper = new TextViewUndoRedo(_hlEditor);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java:191]: The actual fix for the undo redo issue

**Review by gsantner — APPROVED** [review]

**Comment by gsantner** [comment]
Thanks!

**Pull request merged**